### ISSUE TITLE
Expand caching in GitHub Actions workflow

### ISSUE THREAD

**Issue by agitter** [greenelab/covid19-review#305]
**Is your feature request related to a problem? Please describe.**
Recent pull requests have discussed caching some of the literature metadata, like pull request references (#285) and the Allen AI dataset (#286).  Some of the intermediate data could be cached in GitHub Actions.

**Describe the solution you'd like**
Cache intermediate csv files.  Clear the cache daily.

**Describe alternatives you've considered**
The alternative is not caching, which is working fine after #285. This is a low priority issue.
